### ISSUE TITLE
unsafe slash-separated scope (schema-path)

### ISSUE THREAD

**Issue by mirismaili** [eclipsesource/jsonforms#1863]
### Describe the bug

Very similar to #1849 but for `scope`s (or `schemaPath`s).

https://github.com/eclipsesource/jsonforms/blob/075bbc448cfb2bb209e86d6b1ea6460df23a9f64/packages/core/src/generators/uischema.ts#L115-L202

As you can see in the above snippet, `ref` will be `currentRef` of the generated ui-schema:

https://github.com/eclipsesource/jsonforms/blob/075bbc448cfb2bb209e86d6b1ea6460df23a9f64/packages/core/src/generators/uischema.ts#L169-L176

and it will be used as the `scope` of control elements:

https://github.com/eclipsesource/jsonforms/blob/075bbc448cfb2bb209e86d6b1ea6460df23a9f64/packages/core/src/generators/uischema.ts#L195

But it's created by two string-concatenations (_line 162_ & _line 165_):

https://github.com/eclipsesource/jsonforms/blob/075bbc448cfb2bb209e86d6b1ea6460df23a9f64/packages/core/src/generators/uischema.ts#L162-L165

```typescript
const nextRef: string = currentRef + '/properties';
...
const ref = `${nextRef}/${propName}`;
```
The second one is obviously unsafe. Because we don't know anything about `propName`. It may be `foo/bar/baz`. Then we have issues in `toDataPathSegments()`:

https://github.com/eclipsesource/jsonforms/blob/075bbc448cfb2bb209e86d6b1ea6460df23a9f64/packages/core/src/util/path.ts#L58-L68

Because `toDataPathSegments('#/properties/foo/bar/baz')` will be `['foo', 'baz']`!

### Expected behavior

`/` character should safely be escaped or a `string[]` should be used for `scope`s:

For example:

```typescript
// using string[]:
scope = ['#', 'properties', 'foo/bar/baz']

// AJV-like `instancePath` encoding:
scope = '#/properties/foo~1bar~1baz'
```

Then `toDataPathSegments()` can parse its input correctly and will return `['foo/bar/baz']`.

### Steps to reproduce the issue

 1. Go to [Playground](https://mirismaili.github.io/jsonforms-playground)
 2. Set:
    `JSON-Schema`:
    ```json
    {
      "type": "object",
      "properties": {
        "foo/bar": {
          "type": "string"
        }
      }
    }
    ```
    `UI-Schema`:
    ```json
    false
    ```
    `Data`:
    ```json
    {}
    ```

### Screenshots

![image](https://user-images.githubusercontent.com/15800189/147851756-339e6f95-2f8f-43c1-8228-3d270ec15b42.png)

### In which browser are you experiencing the issue?

Google Chrome v96.0.4664.93 (Official Build) (64-bit)

**Comment by sdirix** [comment]
Hi @mirismaili thanks for the report. We are already working on this by encoding slashed whenever necessary, see https://github.com/eclipsesource/jsonforms/pull/1834

**Comment by sdirix** [comment]
This should be fixed now with #1834. Please reopen if you still encounter problems.
